en/docs/chapter_sorting/bubble_sort.md

```diff
@@ -44,16 +44,16 @@ Example code is as follows:
 
 ## Efficiency optimization
 
-We find that if no swaps are performed in a round of "bubbling," the array is already sorted, and we can return the result immediately. Thus, we can add a flag `flag` to monitor this situation and return immediately when it occurs.
+If no swaps occur during a round of "bubbling," the array is already sorted, so we can return immediately. To detect this, we can add a `flag` variable; whenever no swaps are made in a pass, we set the flag and return early.
 
-Even after optimization, the worst-case time complexity and average time complexity of bubble sort remain at $O(n^2)$; however, when the input array is completely ordered, it can achieve the best time complexity of $O(n)$.
+Even with this optimization, the worst-case and average-case time complexities of bubble sort remain $O(n^2)$. However, if the input array is already sorted, the best-case time complexity can be as low as $O(n)$.
```

Comment:
the worst-case and average-case time complexities of bubble sort => the worst time complexity and average time complexity of bubble sort